source/MaterialXGenShader/Nodes/IfNode.cpp

```diff
@@ -47,9 +45,9 @@ void CompareNode::emitFunctionCall(const ShaderNode& node, GenContext& context,
             {
                 shadergen.emitLineBegin(stage);
                 shadergen.emitString("if (", stage);
-                shadergen.emitInput(intest, context, stage);
-                shadergen.emitString(" <= ", stage);
-                shadergen.emitInput(cutoff, context, stage);
+                shadergen.emitInput(value2, context, stage);
```

Comment:
I reversed the value1/value2 order so that the condition string for greater is actually ">" vs the opposite "<=".
Same logic in the end just a bit easier to read instead of reading the negation.